### PR TITLE
refactor: Remove `_dict_update`

### PR DESCRIPTION
**Pull request by hoxbro**
I think this function does exactly what a normal `dict` does if the first argument is a dict.

### CODE REVIEW

File: param/_utils.py [hunk 1]
```diff
@@ -203,15 +203,6 @@ def _is_mutable_container(value):
     return issubclass(type(value), MUTABLE_TYPES)
```
Comment (hoxbro) [inline param/_utils.py:203]: Unrelated but this looks more correct to me
```suggestion
    return isinstance(value, MUTABLE_TYPES)
```

File: param/_utils.py [hunk 2]
```diff
@@ -203,15 +203,6 @@ def _is_mutable_container(value):
...
 
 
-def _dict_update(dictionary, **kwargs):
```
Comment (hoxbro) [inline param/_utils.py:206]: I haven't checked if any downstream uses this.

**Review by maximlt — APPROVED** [review]

**Review by jbednar — APPROVED** [review]
Seems fine to me, though I do always worry if I can't remember why it was done this way originally. Maybe some ancient issue with an earlier Python version?

**Comment by hoxbro** [comment]
> Seems fine to me, though I do always worry if I can't remember why it was done this way originally. Maybe some ancient issue with an earlier Python version?

Either that, it was more complex at some point, or just forgetting that this is possible.

**Pull request merged**